### DISCUSSION TITLE
Sometimes a Timeout Error with send_video()

### DISCUSSION THREAD

**Discussion by joaomino** [MarshalX/atproto#419] [Questions] [answered]
Hi everyone, I’m having an issue with the send_video() function. Occasionally, it works, but most of the time, it throws a Timeout Error (see attached screenshot). It’s not a network issue on my end, as my connection is stable.

Any ideas on what could be causing this?

![sendvideotimeout](https://github.com/user-attachments/assets/e8dc275e-c2b8-402d-9fb0-48ab8161c0d3)

Here’s the code I’m using:

```python
from atproto import Client, models
import os
import time


def main() -> None:
    client = Client()
    client.login(os.environ['BSKY_USERNAME'], os.environ['BSKY_PASSWORD'])

    # replace the path to your video file
    with open('./data/new_video.mp4', 'rb') as f:
        vid_data = f.read()

    root_post_ref = models.create_strong_ref(client.send_video(text='Video posted via send_video() without token', video=vid_data, video_alt='Video posted via send_video - test_game01.mp4'))

    vid_data=None

    time.sleep(60)

    # Reply to the root post. We need to pass ReplyRef with root and parent

        # replace the path to your video file
    with open('./data/test_game01.mp4', 'rb') as f:
        vid_data = f.read()


    reply_to_root = models.create_strong_ref(
        client.send_video(
            text='Reply to the root post via send_post()',
            video=vid_data,
            reply_to=models.AppBskyFeedPost.ReplyRef(parent=root_post_ref, root=root_post_ref),
        )
    )

main()

**Comment by joaomino** [comment]
Could this issue be related to the audio codec? 

I've tried posting the two videos separately. The version that works (test_game01.mp4) uses the AAC audio codec, while the one that throws the error (new_video.mp4) uses MPEG Audio layer 1/2 (mpga). After converting new_video.mp4 to AAC, the error disappeared. Both are using H264 for video.

I'll run a few more tests to confirm-

**Reply by joaomino** [reply]
Not related. 

I converted another video to H264/AAC, and it did not upload. Then I tried again the same video with the H264/mpga pair (new_video.mp4), and it uploaded successfully. 

I updated my code to include a loop with 5 attempts in case the error occurs again.

**Reply by MarshalX** [reply]
Hi! Do you use your own PDS? If your connection is stable as you said, it is probably another client side issue. Do you use macOS? What values of your `ulimit -a`? Did you try to run it on the server?

upd. ah I see this is Windows

**Reply by joaomino** [reply]
Hey! Thanks for getting back to me.

I’m not using a personal PDS; I’m testing this on a local Windows machine.

**Reply by MarshalX** [reply, marked as answer]
Hi, try to just increase timeout. There is not option in `send_video` for it, but you can use upload blob directly and pass own timeout! 

Pseudo code (not ready to copy and paste):
```python
client = Client()
client.login(......)

upload = client.com.atproto.repo.upload_blob(vid_data, timeout=60)
embed = models.AppBskyEmbedVideo.Main(video=upload.blob, alt='my video alt')
client.send_post('blabla my video', embed=embed)
```

**Reply by rankinstudio** [reply]
Could you please explain how to do this with images? I'm also getting timeout errors when posting images.

**Reply by MarshalX** [reply]
> Could you please explain how to do this with images? I'm also getting timeout errors when posting images.

Hi, here is how https://atproto.blue/en/latest/atproto_client/timeouts.html

**Reply by rankinstudio** [reply]
That worked. Thanks

**Reply by IstalriSkolir** [reply]
> > Could you please explain how to do this with images? I'm also getting timeout errors when posting images.
> 
> Hi, here is how https://atproto.blue/en/latest/atproto_client/timeouts.html

I'm having the same issue with client.send_images() and am trying this code without much luck. Importing 'Request' from Atproto doesn't seem to be recogined. I found another post to say import 'Request' directly from the 'requests' module but that led to issues as well
![Screenshot_1](https://github.com/user-attachments/assets/78ea4d77-f054-4ab1-9b29-08b422df70c0)
![Screenshot_2](https://github.com/user-attachments/assets/ef33bec8-de17-4b9f-a6e0-da96435f71a6)

**Reply by MarshalX** [reply]
@IstalriSkolir

1. Update SDK
2. Import request from atproto only

**Reply by IstalriSkolir** [reply]
Yep, updating worked thanks!
